### PR TITLE
ref: abstract class properties aren't a thing

### PR DESCRIPTION
**Pull request by asottile-sentry**

### CODE REVIEW

File: src/sentry/integrations/msteams/webhook.py [hunk 1]
```diff
@@ -143,8 +143,6 @@ def verify_signature(request):
 
 
 class MsTeamsWebhookMixin:
-    provider = "msteams"
```
Comment (asottile-sentry) [inline src/sentry/integrations/msteams/webhook.py:146]: every subclass also sets `provider = "msteams"` so this is redundant

File: src/sentry/integrations/middleware/hybrid_cloud/parser.py [hunk 2]
```diff
@@ -53,22 +52,16 @@ def __init__(
...
-class BaseRequestParser(abc.ABC):
+class BaseRequestParser:
     """Base Class for Integration Request Parsers"""
 
     _METRIC_SUCCESS_KEY = "integrations.middleware.request_parser.success"
     _METRIC_FAILURE_KEY = "integrations.middleware.request_parser.failure"
     _METRICS_INFO_KEY = "integrations.middleware.request_parser.info"
 
-    @property
-    def provider(self) -> str:
```
Comment (asottile-sentry) [inline src/sentry/integrations/middleware/hybrid_cloud/parser.py:64]: mypy (correctly) sees this as an instance property when the intent was classlevel based on usage and implementation

File: src/sentry/integrations/middleware/hybrid_cloud/parser.py [hunk 3]
```diff
@@ -53,22 +52,16 @@ def __init__(
         self.error = error
 
 
-class BaseRequestParser(abc.ABC):
```
Comment (asottile-sentry) [inline src/sentry/integrations/middleware/hybrid_cloud/parser.py:56]: no `@abstractmethod`s so `ABC` doesn't make sense (and just adds runtime overhead)

File: src/sentry/integrations/middleware/hybrid_cloud/parser.py [hunk 4]
```diff
@@ -53,22 +52,16 @@ def __init__(
...
     _METRIC_SUCCESS_KEY = "integrations.middleware.request_parser.success"
     _METRIC_FAILURE_KEY = "integrations.middleware.request_parser.failure"
     _METRICS_INFO_KEY = "integrations.middleware.request_parser.info"
 
-    @property
-    def provider(self) -> str:
-        raise NotImplementedError("'provider' property is required by IntegrationClassification")
-
-    @property
-    def webhook_identifier(self) -> WebhookProviderIdentifier:
```
Comment (evanpurkhiser) [inline src/sentry/integrations/middleware/hybrid_cloud/parser.py:68]: Are we losing doc comments here?

Comment (asottile-sentry) [reply src/sentry/integrations/middleware/hybrid_cloud/parser.py:68]: I didn't really feel like they added any information that they type annotations didn't add 🤷

Comment (asottile-sentry) [reply src/sentry/integrations/middleware/hybrid_cloud/parser.py:68]: (or the resulting stacktrace if they were missing)

**Review by joshuarli — APPROVED** [review]

**Pull request merged**